File: C++/Union of Two Sorted Arrays.cpp
class Solution{
    public:
    vector<int> findUnion(int arr1[], int arr2[], int n, int m)
    {
        set<int> s;
        for(int i = 0;i < n; i++)
            s.insert(arr1[i]);
        for(int j = 0;j < m; j++)
            s.insert(arr2[j]);
        vector<int> ans(s.begin(), s.end());
        return ans;
    }
};

File: JAVA/Union of Two Sorted Arrays.java
class Solution
{
    public static ArrayList<Integer> findUnion(int arr1[], int arr2[], int n, int m)
    {
        ArrayList<Integer> list = new ArrayList<>();
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int num : arr1){
            if(!map.containsKey(num)){
                map.put(num, 1);
                list.add(num);
            }
        }
        for(int num : arr2){
            if(!map.containsKey(num)){
                map.put(num, 1);
                list.add(num);
            }
        }
        Collections.sort(list);
        return list;
    }
}
